From 7d97371413e22c218fba44a53f3d9f3224967bc9 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:05:10 -0400
Subject: [ethernet] qlcnic: Print informational messages only once during driver load

Message-id: <1408626422-3941-27-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88554
O-Subject: [PATCH RHEL 7.1 026/138] qlcnic: Print informational messages only once during driver load.
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Sucheta Chakraborty <sucheta.chakraborty@qlogic.com>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id 9e630955ec01b1a3a5425ede9a17df2f8c8b223b

Signed-off-by: Sucheta Chakraborty <sucheta.chakraborty@qlogic.com>
Signed-off-by: Himanshu Madhani <himanshu.madhani@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic.h b/drivers/net/ethernet/qlogic/qlcnic/qlcnic.h
index 1d5268e..5a32a5c 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic.h
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic.h
@@ -1200,6 +1200,7 @@ struct qlcnic_npar_info {
  u8 promisc_mode;
  u8 offload_flags;
  u8      pci_func;
+ u8      mac[ETH_ALEN];
 };
 
 struct qlcnic_eswitch {
diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c
index a1818da..5d8eb50 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c
@@ -2321,19 +2321,7 @@ int qlcnic_83xx_get_pci_info(struct qlcnic_adapter *adapter,
    i++;
    memcpy(pci_info->mac + sizeof(u32), &cmd.rsp.arg[i], 2);
    i = i + 3;
-   if (ahw->op_mode == QLCNIC_MGMT_FUNC)
-    dev_info(dev, "id = %d active = %d type = %d\n"
-      "\tport = %d min bw = %d max bw = %d\n"
-      "\tmac_addr =  %pM\n", pci_info->id,
-      pci_info->active, pci_info->type,
-      pci_info->default_port,
-      pci_info->tx_min_bw,
-      pci_info->tx_max_bw, pci_info->mac);
   }
-  if (ahw->op_mode == QLCNIC_MGMT_FUNC)
-   dev_info(dev, "Max functions = %d, active functions = %d\n",
-     ahw->max_pci_func, ahw->act_pci_func);
-
  } else {
   dev_err(dev, "Failed to get PCI Info, error = %d\n", err);
   err = -EIO;
diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_vnic.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_vnic.c
index 0248a4c..60a477f 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_vnic.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_vnic.c
@@ -94,13 +94,29 @@ qlcnic_83xx_config_vnic_buff_descriptors(struct qlcnic_adapter *adapter)
  **/
 static int qlcnic_83xx_init_mgmt_vnic(struct qlcnic_adapter *adapter)
 {
- int err = -EIO;
+ struct qlcnic_hardware_context *ahw = adapter->ahw;
+ struct device *dev = &adapter->pdev->dev;
+ struct qlcnic_npar_info *npar;
+ int i, err = -EIO;
 
  qlcnic_83xx_get_minidump_template(adapter);
+
  if (!(adapter->flags & QLCNIC_ADAPTER_INITIALIZED)) {
   if (qlcnic_init_pci_info(adapter))
    return err;
 
+  npar = adapter->npars;
+
+  for (i = 0; i < ahw->act_pci_func; i++, npar++) {
+   dev_info(dev, "id:%d active:%d type:%d port:%d min_bw:%d max_bw:%d mac_addr:%pM\n",
+     npar->pci_func, npar->active, npar->type,
+     npar->phy_port, npar->min_bw, npar->max_bw,
+     npar->mac);
+  }
+
+  dev_info(dev, "Max functions = %d, active functions = %d\n",
+    ahw->max_pci_func, ahw->act_pci_func);
+
   if (qlcnic_83xx_set_vnic_opmode(adapter))
    return err;
 
@@ -115,12 +131,12 @@ static int qlcnic_83xx_init_mgmt_vnic(struct qlcnic_adapter *adapter)
   return err;
 
  qlcnic_83xx_config_vnic_buff_descriptors(adapter);
- adapter->ahw->msix_supported = !!qlcnic_use_msi_x;
+ ahw->msix_supported = qlcnic_use_msi_x ? 1 : 0;
  adapter->flags |= QLCNIC_ADAPTER_INITIALIZED;
  qlcnic_83xx_enable_vnic_mode(adapter, 1);
 
- dev_info(&adapter->pdev->dev, "HAL Version: %d, Management function\n",
-   adapter->ahw->fw_hal_version);
+ dev_info(dev, "HAL Version: %d, Management function\n",
+   ahw->fw_hal_version);
 
  return 0;
 }
diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
index 63b5b03..7ee653b 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
@@ -875,6 +875,7 @@ int qlcnic_init_pci_info(struct qlcnic_adapter *adapter)
   adapter->npars[j].min_bw = pci_info[i].tx_min_bw;
   adapter->npars[j].max_bw = pci_info[i].tx_max_bw;
 
+  memcpy(&adapter->npars[j].mac, &pci_info[i].mac, ETH_ALEN);
   j++;
  }
 
-- 
1.7.1